src/caustic/parameter.py

```diff
@@ -0,0 +1,60 @@
+from typing import Optional
+
+import torch
+from torch import Tensor
+
+__all__ = ("Parameter",)
+
+
+class Parameter:
+    """
+    Represents a static or dynamic parameter. Static parameters are fixed while
+    dynamic parameters must be passed in each time they're required.
+    """
+
+    def __init__(
+        self, value: Optional[Tensor] = None, shape: Optional[tuple[int, ...]] = ()
+    ):
+        # Must assign one of value or shape
+        self._value = value
+        if value is None:
+            if shape is None:
+                raise ValueError("if value is None, a shape must be provided")
+
+            self._shape = shape
+        else:
+            if shape is not None and shape != value.shape:
+                raise ValueError(
+                    f"value's shape {value.shape} does not match provided shape {shape}"
+                )
+
+            self._value = value
+            self._shape = shape
+
+    @property
+    def static(self) -> bool:
```

Comment:
Well I tend to like the idea, makes the code more readable. We pay the price here so that we can replace ```not param.dynamic``` by ```param.static``` in the rest of the code.